### DISCUSSION TITLE
legacy-swc-helpers critical vulnerability

### DISCUSSION THREAD

**Discussion by CarlOfHoly** [swc-project/swc#9400] [General]
According to Github https://github.com/advisories/GHSA-9p8c-m465-cvpr

> Any computer that has this package installed or running should be considered fully compromised. All secrets and keys stored on that computer should be rotated immediately from a different computer. The package should be removed, but as full control of the computer may have been given to an outside entity, there is no guarantee that removing the package will remove all malicious software resulting from installing it.

My project has it as a dependency, but from package-lock I can see that

```
    "node_modules/legacy-swc-helpers": {
      "name": "@swc/helpers",
      "version": "0.4.14",
      "resolved": "https://registry.npmjs.org/@swc/helpers/-/helpers-0.4.14.tgz",
      "integrity": "sha512-4C7nX/dvpzB7za4Ql9K81xK3HPxCpHMgwTZVyf+9JQ6VUbn9jjZVN7/Nkdz/Ugzs2CSjqnL/UPXroiVBVHUWUw==",
      "dependencies": {
        "tslib": "^2.4.0"
      }
    },
```

it resolves to @swc/helpers v 0.4.14. Does that mean that I am safe?

**Comment by magic-akari** [comment]
I suspect that someone recently registered the package "legacy-swc-helpers": https://www.npmjs.com/package/legacy-swc-helpers. However, this name used by "@swc/helpers@0.4.16" is merely a reference to the old swc helpers package and does not actually point to the package on npm. 
So you should not encounter any issues.

**Reply by CarlOfHoly** [reply]
Okay, thank you for your quick response!

**Reply by kdy1** [reply]
Ah, you mean that it's used by `@swc/helpers@0.4.16`.
You are not compromised then.

https://www.npmjs.com/package/@swc/helpers/v/0.4.36?activeTab=code

```json
    "dependencies": {
        "legacy-swc-helpers": "npm:@swc/helpers@=0.4.14",
        "tslib": "^2.4.0"
    },
```

But it would be better if we owned the package name, as some people may try something wrong like depending on directly after looking at package.json

**Reply by magic-akari** [reply]
Indeed, we overlooked this issue. It would be best for us to own it, but it's too late now. 
The package has already been removed, and I guess that also solves the problem.

However, a new concern arises: if our users continue to receive notifications, even if they are false, it can be quite bothersome.
Should we consider renaming it to a package name owned by SWC?

**Reply by kdy1** [reply]
Yeah I think we should do it. How about `@swc/legact-helpers`?

**Reply by samithaf** [reply]
Hey @kdy1 is there any progress with this? we are also literally hitting the same issue and Snyk flagging as a critical issue.

**Reply by kdy1** [reply]
There was no risk unless you ran `npm I legacy-swc-helpers,` and as the npm team took the package down, there was no more threat at all.
In the future we will only use namespaced package names

**Reply by magic-akari** [reply]
> Yeah I think we should do it. How about `@swc/legact-helpers`?

Let's do it.

**Reply by kdy1** [reply]
@magic-akari Do you have the source code? IIRC we didn't commit it to the repository intentionally

**Reply by magic-akari** [reply]
> @magic-akari Do you have the source code? IIRC we didn't commit it to the repository intentionally

We can download it from npm.
It's @swc/helpers@0.4.14.
We won't update it since it's for historical reasons.

~~Or you can just put a single index.js with content：~~


```JavaScript
// @swc/legacy-helpers/index.js
module.exports = require("@swc/hepers");
```

~~and include @swc/helpers@0.4.14 as dependencies.~~

Edited: there are more the one entries. Re-exporting cannot handle it.

**Reply by zachmerrill** [reply]
Any update on a resolution for this? GitHub is still flagging this dependency.

**Reply by kdy1** [reply]
I'll do this right away. Sorry, I forgot about it.

**Reply by kdy1** [reply]
<img width="1840" alt="image" src="https://github.com/user-attachments/assets/42d622a3-24c0-411f-b7b9-9cd745ec47c4">

`@swc/helpers@0.4.37` should have this issue fixed.
